# Preliminary analysis of the Dataset

I just downloaded the dataset and want to determine the structure and the nature of variables in it

In [2]:
import pandas as pd

data = pd.read_json('dataset_airbnb-scraper_2023-06-18_04-11-45-096.json')
data.head(2)

url  \
0  https://www.airbnb.com/rooms/40751388   
1  https://www.airbnb.com/rooms/48645841   

                                                name  stars  numberOfGuests  \
0  Zen 2 bedroom apartment in the heart of El Pob...   4.96               4   
1              Private Jacuzzi Apartment in Provenza   4.64               6   

                         address                  roomType  \
0  Medellín, Antioquia, Colombia  Alojamiento entero: piso   
1  Medellín, Antioquia, Colombia  Alojamiento entero: piso   

                             location  \
0  {'lat': 6.20859, 'lng': -75.56774}   
1  {'lat': 6.20749, 'lng': -75.56672}   

                                             reviews  \
0  [{'author': {'firstName': 'Freddy', 'hasProfil...   
1  [{'author': {'firstName': 'Carlos', 'hasProfil...   

                                             pricing  \
0  {'rate': {'amount': 203, 'amountFormatted': '$...   
1  {'rate': {'amount': 179, 'amountFormatted': '$...   

                                              photos  ...  \
0  [{'caption': '', 'pictureUrl': 'https://a0.mus...  ...   
1  [{'caption': '', 'pictureUrl': 'https://a0.mus...  ...   

                                additionalHouseRules sortedReviews  \
0  - Check in 16:00 PM - Check out 11:00 AM.\n- A...            []   
1  •\tWe are Guest friendly (only one visitor per...            []   

   documentDisplayPictures  sections metadata  \
0                       []        []       {}   
1                       []        []       {}   

                         p3ImpressionId     idStr  \
0  ebcd798b-56b9-245b-9663-269436fe77ef  40751388   
1  fafbee62-1136-4ef7-55cb-49f3b0c97cf6  48645841   

  paidGrowthRemarketingListingIdsStr  \
0                                 []   
1                                 []   

                                       hostGuidebook accessibilityModule  
0                                                NaN                 NaN  
1  {'guidebookUrl': '/s/guidebooks?refinement_pat...                 NaN  

[2 rows x 85 columns]

In [3]:
data.columns

Index(['url', 'name', 'stars', 'numberOfGuests', 'address', 'roomType',
       'location', 'reviews', 'pricing', 'photos', 'primaryHost',
       'additionalHosts', 'isHostedBySuperhost', 'isAvailable', 'calendar',
       'occupancyPercentage', 'listingAmenities', 'rootAmenitySections',
       'seeAllAmenitySections', 'bathroomLabel', 'bedLabel', 'bedroomLabel',
       'hasHouseRules', 'id', 'isNewListing', 'license', 'listingRooms',
       'p3SummaryAddress', 'roomTypeCategory', 'sectionedDescription',
       'tierId', 'guestControls', 'hasSpecialOffer', 'maxNights', 'minNights',
       'priceDetails', 'hasWeWorkLocation', 'isBusinessTravelReady',
       'hometourRooms', 'hometourSections', 'descriptionLocale',
       'initialDescriptionAuthorType', 'localizedCheckInTimeWindow',
       'localizedCheckOutTime', 'city', 'country', 'countryCode',
       'hasHostGuidebook', 'hasLocalAttractions', 'neighborhoodCommunityTags',
       'state', 'paidGrowthRemarketingListingIds', 'hasCommercial

In [4]:
import numpy as np
for c in data.columns:
    if type(data[c][0]) == str:
        print('|'+c+'|str|')
    elif isinstance(data[c][0], np.floating):
        print('|'+c+'|float|')
    elif isinstance(data[c][0], np.int64):
        print('|'+c+'|int|')
    elif type(data[c][0]) == list:
        print('|'+c+'|list|')
    elif type(data[c][0]) == dict:
        print('|'+c+'|dict|')
    elif isinstance(data[c][0], np.bool_):
        print('|'+c+'|bool|')

|url|str|
|name|str|
|stars|float|
|numberOfGuests|int|
|address|str|
|roomType|str|
|location|dict|
|reviews|list|
|pricing|dict|
|photos|list|
|primaryHost|dict|
|additionalHosts|list|
|isHostedBySuperhost|bool|
|isAvailable|float|
|calendar|list|
|occupancyPercentage|float|
|listingAmenities|list|
|rootAmenitySections|list|
|seeAllAmenitySections|list|
|bathroomLabel|str|
|bedLabel|str|
|bedroomLabel|str|
|hasHouseRules|bool|
|id|int|
|isNewListing|bool|
|license|float|
|listingRooms|list|
|p3SummaryAddress|str|
|roomTypeCategory|str|
|sectionedDescription|dict|
|tierId|int|
|guestControls|dict|
|hasSpecialOffer|bool|
|maxNights|int|
|minNights|int|
|priceDetails|list|
|hasWeWorkLocation|bool|
|isBusinessTravelReady|bool|
|hometourRooms|list|
|hometourSections|list|
|descriptionLocale|str|
|initialDescriptionAuthorType|str|
|localizedCheckInTimeWindow|str|
|localizedCheckOutTime|str|
|city|str|
|country|str|
|countryCode|str|
|hasHostGuidebook|bool|
|hasLocalAttractions|bool|
|neigh

In [5]:
occup = data.occupancyPercentage[0]
print('Occupancy as calculated by Apify:', occup)

Occupancy as calculated by Apify: 15.31


In [6]:
## Making sure the occupancyPercentage variable is what I think it is

df = pd.DataFrame.from_dict(data.calendar[0])
#print( df[df.available == True].count() )
oc = df[df.available == False].count() * 100 / df.count()
assert round(occup, 2) == round( oc.available, 2), "Occupancy percentages are incosistent"
print('Occupancy calculated with available days is consistent with the one in table')

Occupancy calculated with available days is consistent with the one in table


## All Variables of full-sized JSON

|Column|Type|Description|
|------|----|-----------|
|url|str||Y|
|name|str||Y|
|stars|float||Y|
|numberOfGuests|int||Y|
|address|str|e.g., 'El Poblado, Medellín, Antioquia, Colombia'|N|
|roomType|str||Y|
|location|dict|e.g., {'lat': 6.20859, 'lng': -75.56774} |Y|
|reviews|list|metadata with reviews information|
|pricing|dict|pricing information, metadata|Y - pricing.rate.amount|
|photos|list| URL of photos|
|primaryHost|dict| info of Host, metadata|
|additionalHosts|list|additional info of Host, metadata|
|isHostedBySuperhost|bool||
|isAvailable|float|1. or nan|
|calendar|list|Every day availability of listing|
|occupancyPercentage|float||
|listingAmenities|list| amenities in the premise|
|rootAmenitySections|list| XXX I do not know|
|seeAllAmenitySections|list||
|bathroomLabel|str|Numero de baños, `'2,5 baños'`|
|bedLabel|str|e.g., `'7 camas'`|
|bedroomLabel|str|e.g., `'3 dormitorios'`|
|hasHouseRules|bool|`True` or `False`|
|id|int|7-8 digit integer|
|isNewListing|bool|
|license|float|i do not know|
|listingRooms|list| type of bed in each room, metadata|
|p3SummaryAddress|str|Seems the same as `address`|
|roomTypeCategory|str| 'entire_home', 'private_room', 'hotel_room', 'shared_room'|
|sectionedDescription|dict|Text, `'authorType', 'description', 'houseRules'...`|
|tierId|int|All Zero, 0|
|guestControls|dict|`'allowsChildren', 'allowsEvents', 'allowsPets', ...`|
|hasSpecialOffer|bool|Mostly `False`|
|maxNights|int||
|minNights|int|
|priceDetails|list|Other expenses `'Gastos de limpieza'` etc.|
|hasWeWorkLocation|bool|All `False`| 
|isBusinessTravelReady|bool|All 'False'|
|hometourRooms|list|???|
|hometourSections|list|???|
|descriptionLocale|str|`'en', 'es', 'it', 'de', ...`|
|initialDescriptionAuthorType|str|All `'human'`| 
|localizedCheckInTimeWindow|str||
|localizedCheckOutTime|str|
|city|str|`'Medellín', 'Amagá', 'El Retiro', 'Envigado', ...`|
|country|str|`'Coloimbia'`|
|countryCode|str|`'Co'`|
|hasHostGuidebook|bool|All `False`|
|hasLocalAttractions|bool|All `False`|
|neighborhoodCommunityTags|list| ????|
|state|str|`'Antioquia', 'Antioquia, Ciudad de Bolivar ', ...`|
|paidGrowthRemarketingListingIds|list| ???|
|hasCommercialHostInfo|bool||
|reviewDetailsInterface|dict|`'reviewCount'` and `'reviewSummary'`|
|reviewsOrder|str|All `'language_country'`|
|nearbyAirportDistanceDescriptions|list|???|
|renderTierId|int|All 0|
|isHotel|bool|
|showReviewTag|bool|All `False`|
|isRepresentativeInventory|bool|???|
|localizedCity|str|`'Medellín', 'Amagá', 'El Retiro', ...`|
|highlights|list|Messages from the visitors|
|highlightsImpressionId|str|something like `'ebcd798b-56b9-245b-9663-269436fe77ef'`|
|categorizedPreviewAmenities|list|
|pointOfInterests|list|Empty ???|
|chinaPointsOfInterestMatcha|dict|Empty >???|
|pageViewType|str|All `'DEFAULT'`|
|previewTags|list|Empty ???|
|seeAllHometourSections|list|Empty ???|
|enableHighlightsVoting|bool|All `False`|
|listingExpectations|list|Some details to take note, e.g., 'noise', 'escaleras'|
|localizedListingExpectations|list|Similar to previous one|
|heroModule|dict|Empty ???|
|reviewsModule|dict|Inside `'localizedOverallRating'`|
|availabilityModule|dict|Empty ???|
|additionalHouseRules|str|Long text, rules|
|sortedReviews|list| Empty ???|
|documentDisplayPictures|list| Empty ???|
|sections|list| Empty ???|
|metadata|dict|Empty ???|
|p3ImpressionId|str| like `ebcd798b-56b9-245b-9663-269436fe77ef`
|idStr|int|Like `40751388`|
|paidGrowthRemarketingListingIdsStr|list|Empty ???|

In [52]:
print( data.pricing[0]['rate']['amount'] )
print( data.listingRooms[0][0]['beds'][0]['type'] )
print( data.sectionedDescription[0]['locale'])
print( len( data.sectionedDescription[0]['description'] ) )
print( data.priceDetails[0][0]['attributeType'], data.priceDetails[0][0]['value'])
print( data.reviewDetailsInterface[0]['reviewCount'])
print( 'Balcon? ', 'balc' in data.categorizedPreviewAmenities[0][1]['amenities'])
print( len(data.additionalHouseRules[0]) )

203
queen_bed
en
1464
CLEANING_FEE 69 €
105
Balcon?  True
881


In [51]:
data.heroModule

0       {'categorizedPhotos': []}
1       {'categorizedPhotos': []}
2       {'categorizedPhotos': []}
3       {'categorizedPhotos': []}
4       {'categorizedPhotos': []}
                  ...            
1993    {'categorizedPhotos': []}
1994    {'categorizedPhotos': []}
1995    {'categorizedPhotos': []}
1996    {'categorizedPhotos': []}
1997    {'categorizedPhotos': []}
Name: heroModule, Length: 1998, dtype: object

In [53]:
df = data.copy()

df.drop(columns=['address', 'reviews', 'photos', 'additionalHosts', 'rootAmenitySections'], inplace=True)
df.drop(columns=['seeAllAmenitySections', 'license', 'p3SummaryAddress', 'tierId', 'hasSpecialOffer'], inplace=True)
df.drop(columns=['hasWeWorkLocation', 'isBusinessTravelReady', 'hometourRooms', 'hometourSections'], inplace=True)
df.drop(columns=['country', 'countryCode', 'hasHostGuidebook', 'hasLocalAttractions', 'neighborhoodCommunityTags'], inplace=True)
df.drop(columns=['state', 'paidGrowthRemarketingListingIds', 'reviewsOrder', 'nearbyAirportDistanceDescriptions'], inplace=True)
df.drop(columns=['renderTierId', 'showReviewTag', 'localizedCity', 'highlights', 'highlightsImpressionId'], inplace=True)
df.drop(columns=['pointOfInterests', 'chinaPointsOfInterestMatcha', 'pageViewType', 'previewTags'], inplace=True)
df.drop(columns=['seeAllHometourSections', 'enableHighlightsVoting', 'listingExpectations'], inplace=True)
df.drop(columns=['localizedListingExpectations', 'heroModule', 'reviewsModule', 'availabilityModule'], inplace=True)
df.drop(columns=['sortedReviews', 'documentDisplayPictures', 'sections', 'metadata', 'idStr'])




url  \
0     https://www.airbnb.com/rooms/40751388   
1     https://www.airbnb.com/rooms/48645841   
2     https://www.airbnb.com/rooms/49654374   
3     https://www.airbnb.com/rooms/51173424   
4     https://www.airbnb.com/rooms/40288291   
...                                     ...   
1993  https://www.airbnb.com/rooms/45115057   
1994  https://www.airbnb.com/rooms/23709640   
1995  https://www.airbnb.com/rooms/34904012   
1996   https://www.airbnb.com/rooms/5401517   
1997  https://www.airbnb.com/rooms/45242333   

                                                   name  stars  \
0     Zen 2 bedroom apartment in the heart of El Pob...   4.96   
1                 Private Jacuzzi Apartment in Provenza   4.64   
2        Finca Los Sauces Moderna y Llena De Naturaleza    NaN   
3                  Espectacular 4 Hab cerca en Provenza   5.00   
4     Gaia House (Laureles) 6 Habit'+Suit&enorme ter...   5.00   
...                                                 ...    ...   
1993  Apartaestudio cerca al metro|Poblado|Equipado ...   4.75   
1994      unique location aparment near metro station 4   4.57   
1995            El Poblado, Zona Rosa/Residencial.103v2   4.47   
1996                       Lovely 1 BR Flat in Medellín   4.86   
1997           3201 confortable apartamento en laureles   4.59   

      numberOfGuests                                       roomType  \
0                  4                       Alojamiento entero: piso   
1                  6                       Alojamiento entero: piso   
2                 16                      Alojamiento entero: villa   
3                  8                       Alojamiento entero: piso   
4                 14                   Alojamiento entero: vivienda   
...              ...                                            ...   
1993               3  Alojamiento entero: apartamento con servicios   
1994               2                       Alojamiento entero: piso   
1995               4                       Alojamiento entero: piso   
1996               2                       Alojamiento entero: piso   
1997               2                       Alojamiento entero: piso   

                                location  \
0     {'lat': 6.20859, 'lng': -75.56774}   
1     {'lat': 6.20749, 'lng': -75.56672}   
2     {'lat': 6.02727, 'lng': -75.70012}   
3     {'lat': 6.20688, 'lng': -75.56405}   
4         {'lat': 6.247, 'lng': -75.589}   
...                                  ...   
1993  {'lat': 6.19398, 'lng': -75.58002}   
1994   {'lat': 6.2571, 'lng': -75.58427}   
1995      {'lat': 6.209, 'lng': -75.566}   
1996  {'lat': 6.19025, 'lng': -75.58108}   
1997   {'lat': 6.24556, 'lng': -75.5941}   

                                                pricing  \
0     {'rate': {'amount': 203, 'amountFormatted': '$...   
1     {'rate': {'amount': 179, 'amountFormatted': '$...   
2     {'rate': {'amount': 421, 'amount_formatted': '...   
3     {'rate': {'amount': 178, 'amountFormatted': '$...   
4     {'rate': {'amount': 200, 'amountFormatted': '$...   
...                                                 ...   
1993  {'rate': {'amount': 37, 'amountFormatted': '$3...   
1994  {'rate': {'amount': 30, 'amountFormatted': '$3...   
1995  {'rate': {'amount': 27, 'amountFormatted': '$2...   
1996  {'rate': {'amount': 28, 'amountFormatted': '$2...   
1997  {'rate': {'amount': 26, 'amountFormatted': '$2...   

                                            primaryHost  isHostedBySuperhost  \
0     {'id': 270110438, 'firstName': 'Derek', 'isSup...                 True   
1     {'id': 238034915, 'firstName': 'HostinGroup', ...                False   
2     {'id': 110489568, 'firstName': 'Laura', 'isSup...                False   
3     {'id': 393284767, 'firstName': 'MOX Rentals', ...                False   
4     {'id': 240570191, 'firstName': 'Aleluya Hilel'...                 True   
...                                                 ...                  ...   
1993  {'id': 364352534, 'firstN